src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  color: Color,
+  shape: Shape,
 
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((x, y) => y - x);
 
-export class Circle {
+    if ((sides[0] >= sides[1] + sides[2])
+        || a < 1 || b < 1 || c < 1) {
+      throw new Error('All sides must be greater than 0');
```

Comment:
we have a condition that all sides must be greater than 0 , and we can have side = 1
so we should compare like a < 0 or a <= 1